src/gz21_ocean_momentum/cli/train.py

```diff
@@ -0,0 +1,165 @@
+
+import gz21_ocean_momentum.common.cli as cli
+import gz21_ocean_momentum.common.assorted as common
+import gz21_ocean_momentum.common.bounding_box as bounding_box
+import gz21_ocean_momentum.lib.model as lib
+import gz21_ocean_momentum.models.submodels as submodels
+import gz21_ocean_momentum.models.transforms as transforms
+import gz21_ocean_momentum.models.models1 as model
+import gz21_ocean_momentum.train.losses as loss
+from gz21_ocean_momentum.train.base import Trainer
+from gz21_ocean_momentum.inference.metrics import MSEMetric, MaxMetric
+from gz21_ocean_momentum.data.datasets import Subset_, ConcatDataset_
+
+import configargparse
+
+import os
+
+import xarray as xr
+import numpy as np
+
+import torch
+from torch.utils.data import DataLoader, ConcatDataset
+from torch import optim
+from torch.optim.lr_scheduler import MultiStepLR
+
+# TODO probably temporary
+import tempfile
+
+# TODO ideally temporary but probably not
+import copy
+
+_cli_desc = """
+Train a Pytorch neural net to predict subgrid ocean momentum forcing from
+ocean surface velocity.
+
+Uses data generated by the GZ21 data step script.
+"""
+
+p = configargparse.ArgParser(description=_cli_desc)
+p.add("--config-file", is_config_file=True, help="config file path")
+p.add("--in-train-data-dir",         type=str,   required=True, help="training data in zarr format, containing ocean velocities and forcings")
+p.add("--subdomains-file",           type=str,   required=True, help="YAML file describing subdomains to split input data into (see readme for format)")
+p.add("--batch-size",                 type=int,   required=True, help="TODO")
+p.add("--epochs",                    type=int,   required=True, help="number of epochs to train for")
+p.add("--out-model",                 type=str,   required=True, help="save trained model to this path")
+p.add("--initial-learning-rate",     type=float, required=True, help="initial learning rate for optimization algorithm")
+p.add("--decay-factor",              type=float, required=True, help="learning rate decay factor, applied each time an epoch milestone is reached")
+p.add("--decay-at-epoch-milestones", type=int, action="append", required=True, help="milestones to decay at. May specify multiple times. Must be strictly increasing with no duplicates")
+p.add("--device",                    type=str, default="cuda:0", help="neural net device (e.g. cuda:0, cpu)")
+p.add("--weight-decay",              type=float, default=0.0, help="Weight decay parameter for Adam loss function. Deprecated, default 0.")
+p.add("--train-split-end",  type=float, required=True, help="0>=x>=1. Use 0->x of input dataset for training")
+p.add("--test-split-start", type=float, required=True, help="0>=x>=1. Use x->end of input dataset for testing. Must be greater than --train-split-start")
+p.add("--printevery", type=int, default=20)
+options = p.parse_args()
+
+# TODO raehik 2023-11-13: parse, don't validate
```

Comment:
Change to comment not a TODO unless it requires a new issue.

Reply:
This was a small gripe, not really an issue -- removed note.